service/src/main/java/org/jboss/sbomer/service/feature/sbom/features/generator/rpm/controller/BrewRPMController.java

```diff
@@ -0,0 +1,289 @@
+import io.javaoperatorsdk.operator.api.reconciler.ControllerConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.Dependent;
+import jakarta.validation.ValidationException;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * <p>
+ * Reconciler working on the {@link GenerationRequest} entity and the {@link GenerationRequestType#RPM} type.
+ * </p>
+ *
+ * <p>
+ * This reconciler acts only on resources marked with following labels (all of them must exist on the resource):
+ *
+ * <ul>
+ * <li>{@code app.kubernetes.io/part-of=sbomer}</li>
+ * <li>{@code app.kubernetes.io/component=sbom}</li>
+ * <li>{@code app.kubernetes.io/managed-by=sbom}</li>
+ * <li>{@code sbomer.jboss.org/generation-request}</li>
+ * <li>{@code sbomer.jboss.org/generation-request-type}</li>
+ * </ul>
+ * </p>
+ */
+@ControllerConfiguration(
+        labelSelector = "app.kubernetes.io/part-of=sbomer,app.kubernetes.io/component=sbom,app.kubernetes.io/managed-by=sbom,sbomer.jboss.org/type=generation-request,sbomer.jboss.org/generation-request-type=brew-rpm",
+        namespaces = { Constants.WATCH_CURRENT_NAMESPACE },
+        dependents = { @Dependent(
+                type = TaskRunBrewRPMGenerateDependentResource.class,
+                useEventSourceWithName = EVENT_SOURCE_NAME) })
+@Slf4j
+public class BrewRPMController extends AbstractController {
+
+    @Override
+    protected UpdateControl<GenerationRequest> updateRequest(
+            GenerationRequest generationRequest,
+            SbomGenerationStatus status,
+            GenerationResult result,
+            String reason,
+            Object... params) {
+
+        if (generationRequest.getStatus() != null) {
+            String label = switch (generationRequest.getStatus()) {
+                case GENERATING -> SbomGenerationPhase.GENERATE.name().toLowerCase();
+                default -> null;
+            };
+
+            if (label != null) {
+                generationRequest.getMetadata().getLabels().put(Labels.LABEL_PHASE, label);
+            }
+        }
+
+        generationRequest.setStatus(status);
+        generationRequest.setResult(result);
+        generationRequest.setReason(MessageFormatter.arrayFormat(reason, params).getMessage());
+        return UpdateControl.updateResource(generationRequest);
+    }
+
+    /**
+     * <p>
+     * Handles updates to {@link GenerationRequest} being in progress.
+     * </p>
+     *
+     * @param generationRequest
+     * @param secondaryResources
+     * @return
+     */
+    @Override
+    protected UpdateControl<GenerationRequest> reconcileGenerating(
+            GenerationRequest generationRequest,
+            Set<TaskRun> secondaryResources) {
+
+        log.debug("Reconcile GENERATING for '{}'...", generationRequest.getName());
+
+        TaskRun generateTaskRun = findTaskRun(secondaryResources, SbomGenerationPhase.GENERATE);
```

Comment:
I imagine this is a first pass, right? I imagine there will be more than one TaskRun, because we will have as many TaskRuns as many Product Versions in the advisory.